src/hooks/fix-decompress.ts

```diff
@@ -0,0 +1,93 @@
+import zlib from 'zlib';
+import { ClientRequest, IncomingMessage } from 'http';
+import { HandlerFunction } from 'got-cjs';
+import { PassThrough, Transform } from 'stream';
+import mimicResponse from 'mimic-response';
+
+const onResponse = (response: IncomingMessage, propagate: (fixedResponse: IncomingMessage) => void) => {
```

Comment:
so propagate is just a plain callback?

Reply:
Yep. This could be written as async/await, but in this case would be much uglier.

Reply:
Yeah, no probs. I was just confused by the name. Thought it's something different than a callback, since it's named differently.

Reply:
I named it this way because it emits the `response` event. `callback` would do too but is more general.